src/EPR.Calculator.Frontend/Controllers/CalculationRunNameController.cs

```diff
@@ -1,13 +1,44 @@
 ﻿using EPR.Calculator.Frontend.Constants;
+using EPR.Calculator.Frontend.Models;
 using Microsoft.AspNetCore.Mvc;
 
 namespace EPR.Calculator.Frontend.Controllers
 {
     public class CalculationRunNameController : Controller
     {
+        private const string CalculationRunNameIndexView = ViewNames.CalculationRunNameIndex;
+        private const string CalculationRunConfirmationAction = "Index";
+        private const string CalculationRunConfirmationController = "CalculationRunConfirmation";
+
         public IActionResult Index()
         {
-            return this.View(ViewNames.CalculationRunNameIndex);
+            return this.View(CalculationRunNameIndexView);
+        }
+
+        [HttpPost]
+        public IActionResult RunCalculator(string calculationName)
+        {
+            if (IsCalculationNameInvalid(calculationName))
```

Comment:
Do we need a method for this?
You could say `if(string.IsNullOrEmpty(calculationName)) 
{
// code here
}`

Reply:
@udaydefra  That should be fine for now but I kept for a reason because I need to do other validations as part upcoming tickets I just put a placeholder and will do the method name changes as per validations later.

Reply:
We will do it when the integration starts

Reply:
I have changed for you